### PR TITLE
Cleanup after Ansible runner integration

### PR DESCRIPTION
**Pull request by lfu**
Depends on https://github.com/ManageIQ/manageiq-automation_engine/pull/383.
Related to https://github.com/ManageIQ/manageiq/pull/19383.

https://bugzilla.redhat.com/show_bug.cgi?id=1765682

@miq-bot add_label refactoring, changelog/yes, ivanchuk/yes, hammer/no, embedded ansible, services

### CODE REVIEW

File: content/automate/ManageIQ/AutomationManagement/AnsibleTower/Operations/Methods.class/__methods__/available_credentials.rb [hunk 1]
```diff
@@ -25,7 +25,7 @@ def provider
               # for reconfig and retire use service
               # Or use the embedded_ansible_provider
               service = @handle.root['service_template'] || @handle.root['service']
-              service.try(:job_template, 'Provision').try(:manager) || embedded_ansible_provider
+              service.try(:playbook, 'Provision').try(:manager) || embedded_ansible_provider
```
Comment (gmcculloug) [inline content/automate/ManageIQ/AutomationManagement/AnsibleTower/Operations/Methods.class/__methods__/available_credentials.rb]: @lfu I have come around to agreeing with you on this change after a bit more searching.  The only model that supported a `job_template` method **that also accepts a parameter** was `ServiceTemplateAnsiblePlaybook`

With the changes to support Runner and the related refactorings, that method was renamed to `playbook`.  See https://github.com/ManageIQ/manageiq/blob/master/app/models/service_template_ansible_playbook.rb#L75

I also checked the code base at the time this `available_credentials.rb` script was initially created `git checkout 'master@{2017-02-16}'` and see that `ServiceTemplateAnsiblePlaybook` was the only model to support `job_template` with a parameter.

File: spec/content/automate/ManageIQ/AutomationManagement/AnsibleTower/Operations/Methods.class/__methods__/available_credentials_spec.rb [hunk 2]
```diff
@@ -2,8 +2,8 @@
 
 describe ManageIQ::Automate::AutomationManagement::AnsibleTower::Operations::AvailableCredentials do
   let(:ansible_manager) { FactoryBot.create(:automation_manager_ansible_tower) }
```
Comment (gmcculloug) [inline spec/content/automate/ManageIQ/AutomationManagement/AnsibleTower/Operations/Methods.class/__methods__/available_credentials_spec.rb]: @lfu Should this now be `embedded_automation_manager_ansible` or `provider_embedded_ansible`?

File: content/automate/ManageIQ/AutomationManagement/AnsibleTower/Operations/Methods.class/__methods__/available_credentials.rb (hunk 1 above)
Comment (mkanoor) [reply content/automate/ManageIQ/AutomationManagement/AnsibleTower/Operations/Methods.class/__methods__/available_credentials.rb]: 👍

**Comment by simaishi** [comment]
Ivanchuk backport details:

```
$ git log -1
commit f20f66725193b2647a34235537de31af468584e5
Author: tina <tfitzger@redhat.com>
Date:   Thu Oct 31 10:09:30 2019 -0400

    Merge pull request #594 from lfu/ansible_runner_cleanup_credential

    Cleanup after Ansible runner integration

    (cherry picked from commit 61527d22c871b56b969fcb6027acccc70439c53c)

    https://bugzilla.redhat.com/show_bug.cgi?id=1784103
```

**Pull request merged**